apps/jobs/sender.py

```diff
@@ -72,10 +74,39 @@ def publish_submission_message(message):
         challenge = Challenge.objects.get(pk=message["challenge_pk"])
     except Challenge.DoesNotExist:
         logger.exception(
-            "Challenge does not exist for the given id {}".format(message["challenge_pk"])
+            "Challenge does not exist for the given id {}".format(
+                message["challenge_pk"]
+            )
         )
         return
     queue_name = challenge.queue
+    slack_url = challenge.slack_webhook_url
     queue = get_or_create_sqs_queue(queue_name)
     response = queue.send_message(MessageBody=json.dumps(message))
+    # sending slack notification
+    if slack_url:
+        challenge_name = challenge.title
+        phase_name = ChallengePhase.objects.get(pk=message["phase_pk"]).name
```

Comment:
@Ayukha you can get the `phase_name` from the Submission object itself. There is no need for this query.